### ISSUE TITLE
No matching distribution found for scann

### ISSUE THREAD

**Issue by tankwin08** [google-research/google-research#779]
I try to install scann using _`pip install scann`_, it give me the following error:

_ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann_

Here is package version in my env:

**tensorflow=2.6.0
tensorflow recommender=v0.5.2
tensorflow dataset=4.4.0**

It can install the scann on Colab and I used the same env in my local computer, BUT it did not work. Any suggestions are appreciated.

**Comment by xdwang0726** [comment]
Same here! I got the same error message, but my backend is Pytorch 3.7.

**Comment by cramraj8** [comment]
Found any solutions ?

**Comment by tansaku** [comment]
same issue for me

**Comment by kim-sardine** [comment]
it says `ScaNN supports Linux environments running Python versions 3.6-3.9`
I found it works on docker with python 3.6-3.9

**Comment by tansaku** [comment]
that's good to hear @kim-sardine - can you be more specific about which version works for you?  3.8.7 not working for me:

```
❯ python
Python 3.8.7 (default, Feb  3 2021, 17:06:40) 
[Clang 12.0.0 (clang-1200.0.32.28)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
❯ pip install scann
ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann
```

or you mean you've tried with all versions between 3.6 and 3.9 and they all work?

**Comment by XiaoConstantine** [comment]
Try:
```bash
export SYSTEM_VERSION_COMPAT=1
```
and then proceed

**Comment by tansaku** [comment]
thanks @XiaoConstantine - however that does not appear to have helped:

```
. /opt/anaconda3/bin/activate && conda activate /opt/anaconda3; 
❯ . /opt/anaconda3/bin/activate && conda activate /opt/anaconda3;
❯ export SYSTEM_VERSION_COMPAT=1
❯ pip install scann
ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann
```

**Comment by tansaku** [comment]
here are the current versions I have:

```
tensorflow                         2.4.1
tensorflow-datasets                4.4.0
tensorflow-estimator               2.4.0
tensorflow-hub                     0.12.0
tensorflow-metadata                0.29.0
tensorflow-recommenders            0.5.1
```

**Comment by XiaoConstantine** [comment]
@tansaku I see, here's what I suggest you try:
Reinstall python@3.8, note: python version need to be greater than 3.8.10
```
brew reinstall python@3.8
```
(if you use homebrew, you should end up with 3.8.12)
Make sure your virtual env/any kind of env management system use python3.8.12 as your virtualenv python interpreter and then pip install scann

reference: https://bugs.python.org/issue41100

**Comment by Qucy** [comment]
python 3.9.7 - April 16, 2022 still got this issue
pip install -q scann
ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann

**Comment by jinseriouspark** [comment]
have you every check pip version? 
I tried upgrade and solved it 
pip install -U pip

**Comment by satheesh-anecure** [comment]
Same with me

**Comment by BILLXZY1215** [comment]
Same, any workaround?

**Comment by leemengtaiwan** [comment]
Same here with 
- `Python 3.9.13`
- `pip-22.2.2`
- `tensorflow==2.9.2`
- `tensorflow-recommenders==0.7.0`

```sh
❯ pip install scann
ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann
```

😢

**Comment by KobeAmerijckxPT** [comment]
yup, same here

**Comment by OKrawi** [comment]
I'm facing the same problem. Does anyone have a solution?

**Comment by kmcmearty** [comment]
Same problem. Python 3.9.12

**Comment by XiaoConstantine** [comment]
There's no binary distribution for macos, if you need scann on mac, feel free to try out instructions in this PR: https://github.com/google-research/google-research/pull/830 (Build from source on mac), once you finished all the steps, you are good to go.

**Comment by christian33moreno** [comment]
Still having the issue...any solutions found??

**Comment by MakC-Ukr** [comment]
Same issue, can't find a solution

**Comment by nmartin-sens** [comment]
Same issue. Python 3.8.5

**Comment by austa-ias** [comment]
Same issue here ! Cannot install scann on python3.8

**Comment by ZFbaby** [comment]
No matching distribution found for scann==1.2.6 (from tflite-model-maker)
I have the same problem, can anyone help me?

**Comment by ZFbaby** [comment]
> I try to install scann using _`pip install scann`_, it give me the following error:
> 
> _ERROR: Could not find a version that satisfies the requirement scann (from versions: none) ERROR: No matching distribution found for scann_
> 
> Here is package version in my env:
> 
> **tensorflow=2.6.0 tensorflow recommender=v0.5.2 tensorflow dataset=4.4.0**
> 
> It can install the scann on Colab and I used the same env in my local computer, BUT it did not work. Any suggestions are appreciated.

Did you solve this problem?

**Comment by BaraaZ95** [comment]
scann works when you install it via wsl. It doesn't seem to have windows support atm

**Comment by nmartin-sens** [comment]
I've found a solution to install it manually. See: https://github.com/CompVis/latent-diffusion/issues/147#issuecomment-1279934409

**Comment by RubensZimbres** [comment]
This versions work perfectly for me:

`Successfully installed flatbuffers-1.12 keras-2.9.0 scann-1.2.7 tensorboard-2.9.1 tensorflow-2.9.3 tensorflow-estimator-2.9.0`

I uninstalled tensorflow-gpu and installed scann==1.2.7 via pip

My versions of Tensorflow Recommenders and Tensorflow Datasets are 0.7.0 and 4.7.0, respectively

**Comment by Srolita2021** [comment]
I faced the same problem with  Python  from 3.7 and 3.8 , pip 19  the matter has something to do with pip ,the solution was to upgrade pip
pip install --upgrade pip

**Comment by HripsimeS** [comment]
Hello. I am facing the same error message with **pip install scann** on my Windows 11.
ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann

Did anyone found a solution how to install properly **scann** for Windows? Thanks in advance!

**Comment by hakimus2k** [comment]
Hi! 
I have the same problem. I was trying to install scann as requirement for tflite-model-maker under enviroments with python 3.10, 3.19, 3.8 in windows 10. 
I read that scann is a linux lib. Ok, I tried to install from my hosted linux server which doesnt many libraries installed. The result was exactly the as as in windows:
**ERROR: Could not find a version that satisfies the requirement scann==1.2.6 (from versions: none)
ERROR: No matching distribution found for scann==1.2.6**

I read that it works in WSL, great, I tried to install scann from a virtual env on Cygwin. Same result.
Bottom line: pip cannot fint scann netiher from Windows, Cygwin nor Linux server.

I dont know how to check if the lib is available, or maybe which kind of requirement do I need to install.

Any light to this poor lost soul?

**Comment by tamnn25** [comment]
Same issue in 2023 with python 3.7, 3.8, 3.9. Please help me I'm using Mac m1.
ERROR: Could not find a version that satisfies the requirement tensorflow-cpu==2.8.1 (from versions: none)
ERROR: No matching distribution found for tensorflow-cpu==2.8.1

**Comment by pdwytr** [comment]
Facing the same error on mac m1

**Comment by newbie-lad** [comment]
Modern “software development” 😂
There is a solution to all your problems, boys...

**Comment by iozyigit0** [comment]
Not sure why google like to push people away from their product. This issue let me spent lot of time for nothing just to install this om M1 chip :)

**Comment by simondoesstuff** [comment]
> Not sure why google like to push people away from their product. This issue let me spent lot of time for nothing just to install this om M1 chip 👎

but, you **did** manager to install it on the m1?

**Comment by iozyigit0** [comment]
> 

@simondoesstuff
Unfortunately no, I gave up.

**Comment by ZhuohanYU** [comment]
Hi I got same issue here on M1 chip.

**Comment by Tiago4D** [comment]
same here

**Comment by cikichen** [comment]
same issue on m1

**Comment by mohcineelharras** [comment]
same issue here

**Comment by saskiabosma** [comment]
I had the same Scann install issue for inference testing (my TFRS+Scann model was not trained on M1), it can be solved by using docker. Start colima with the following options: `colima start --arch x86_64 --cpu 4 --cpu-type "max" --memory 8`. Then run the image with your model (see example [here](https://www.tensorflow.org/tfx/serving/docker#creating_your_own_serving_image), replacing the TFServing image by the [TF+Scann image](https://hub.docker.com/r/google/tf-serving-scann)): `docker run --platform linux/amd64 -p 8501:8501 -t yourimage:yourtag`.

**Comment by abhishek0093** [comment]
same issue on m1. please provide support for mac. 
`ERROR: Could not find a version that satisfies the requirement scann (from versions: none)
ERROR: No matching distribution found for scann`

**Comment by milind29** [comment]
same issue with python version 3.10

**Comment by pirevi** [comment]
2 years and no solution. thanks google!

**Comment by YaroslavKharchenko0** [comment]
If you on mac os you can try 
` pip install git+https://github.com/google-research/google-research.git#subdirectory=scann`

**Comment by Sanicey** [comment]
read this url: https://pypi.org/project/scann/1.3.1/   it seems that ScaNN only supports Linux environments running Python versions 3.9-3.12

**Comment by mlewelt** [comment]
That would mean you cannot use tflite-model-maker on windows? No way.

**Comment by jon1enforce** [comment]
everything is wrong with this world :dancers: 

build from old source?

python configure.py
CC=clang-17 bazel build -c opt --features=thin_lto --copt=-mavx --copt=-mfma --cxxopt="-std=c++17" --copt=-fsized-deallocation --copt=-w :build_pip_pkg
./bazel-bin/build_pip_pkg

**Comment by SPMT01** [comment]
Python 3.8.20；pip 24.3.1
Same problem. Has anyone solved it?

**Comment by chrisjune** [comment]
> Python 3.8.20；pip 24.3.1 Same problem. Has anyone solved it?

I manually installed the newest version of scann using git
"""
pip install git+https://github.com/google-research/google-research.git#subdirectory=scann
"""

**Comment by somwrks** [comment]
> > Python 3.8.20；pip 24.3.1 Same problem. Has anyone solved it?
> 
> I manually installed the newest version of scann using git """ pip install git+https://github.com/google-research/google-research.git#subdirectory=scann """

does it work then?

**Comment by chrisjune** [comment]
> does it work then?

I could install newer version of ScaNN, but it woudn't work. 
So I installed python3.10, now I can newer version of it.
I added Dockerfile and requirements.txt

## Dockerfile
```Dockerfile
FROM {BaseImage}

# Prerequisite
RUN apt update
RUN apt -y install software-properties-common
RUN add-apt-repository ppa:deadsnakes/ppa
RUN apt update

# Install python
RUN apt install -y python3.10 python3-pip

# Copy source code
WORKDIR /app

COPY requirements.txt .
RUN pip3 install --upgrade pip
RUN pip3 install --no-cache-dir -r requirements.txt

COPY src /app/src
ENV PYTHONPATH /app/src
```

## requirements.txt
```txt
scann==1.3.0
```

**Comment by somwrks** [comment]
awesome! thanks

**Comment by rising-dancho** [comment]
i tried running it in virtual box with ubuntu installed. it worked that way as a workaround.

**Comment by stampixel** [comment]
> i tried running it in virtual box with ubuntu installed. it worked that way, as a workaround.

not smart. bad idea.
